spec/models/post_spec.rb

```diff
@@ -35,5 +35,16 @@
         expect(post.up_votes).to eq(1)
       end
     end
+
+    describe "#save_with_initial_vote" do
+      it "saves a post only if the initial vote saves as well" do
+        post = described_class.new(title: "Check One Two",
+                        body:  "Sibilance. Sibilance.",
+                        user:  authenticated_user,
+                        created_at: Time.new)
```

Comment:
To actually answer the question :stuck_out_tongue:, I would have done something like this:

```ruby
let(:post) { associated_post_unsaved } # Or some other factory method (e.g. Fabrication gem)
...
describe "#save_with_initial_vote" do
  it "saves a post only if the initial vote saves as well" do
    expect(post.save_with_initial_vote).to eq true
    expect(post).to be_persisted # comes after the line above where we call save_with_initial_vote
  end

  it "associates a Vote object with the post" do
    post.save_with_initial_vote
    expect(post.vote).to be_a Vote # or .to_not be_nil
    expect(post.vote).to be_persisted
  end
end
```

You could also test if saving the post or saving the vote fails. In that case post should not be persisted, and vote may or may not be `nil` but should not be persisted if it's not nil. You will need a factory method with an invalid post to test that.